[2024-11-05T21:58:02.799+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_transform.transform manual__2024-11-05T21:58:02.135834+00:00 [queued]>
[2024-11-05T21:58:02.811+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_transform.transform manual__2024-11-05T21:58:02.135834+00:00 [queued]>
[2024-11-05T21:58:02.812+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-11-05T21:58:02.824+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): transform> on 2024-11-05 21:58:02.135834+00:00
[2024-11-05T21:58:02.832+0000] {standard_task_runner.py:57} INFO - Started process 1569 to run task
[2024-11-05T21:58:02.835+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_transform', 'transform', 'manual__2024-11-05T21:58:02.135834+00:00', '--job-id', '60', '--raw', '--subdir', 'DAGS_FOLDER/spark_transform.py', '--cfg-path', '/tmp/tmp04l9uznd']
[2024-11-05T21:58:02.836+0000] {standard_task_runner.py:85} INFO - Job 60: Subtask transform
[2024-11-05T21:58:02.904+0000] {task_command.py:416} INFO - Running <TaskInstance: spark_transform.transform manual__2024-11-05T21:58:02.135834+00:00 [running]> on host c5156f201d33
[2024-11-05T21:58:02.983+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_transform' AIRFLOW_CTX_TASK_ID='transform' AIRFLOW_CTX_EXECUTION_DATE='2024-11-05T21:58:02.135834+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-05T21:58:02.135834+00:00'
[2024-11-05T21:58:02.995+0000] {base.py:73} INFO - Using connection ID 'SPARK_CONNECTION' for task execution.
[2024-11-05T21:58:02.996+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name arrow-spark jobs/example.py
[2024-11-05T21:58:03.102+0000] {spark_submit.py:536} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-11-05T21:58:03.103+0000] {spark_submit.py:536} INFO - JAVA_HOME is not set
[2024-11-05T21:58:03.129+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark jobs/example.py. Error code is: 1.
[2024-11-05T21:58:03.133+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=spark_transform, task_id=transform, execution_date=20241105T215802, start_date=20241105T215802, end_date=20241105T215803
[2024-11-05T21:58:03.140+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 60 for task transform (Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark jobs/example.py. Error code is: 1.; 1569)
[2024-11-05T21:58:03.168+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-11-05T21:58:03.187+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
